nats.go

```diff
@@ -492,6 +493,10 @@ type Subscription struct {
 	// only be processed by one member of the group.
 	Queue string
 
+	// ConsumerConfig is the configuration for the JetStream consumer if one was created
+	// or updated using the subscription options
+	ConsumerConfig *ConsumerConfig
```

Comment:
When we have some other feature we don't know about, that is not 'consumer' specific this will require a breaking change in the signature. I think the function signatures should have a "SubscriptionOption" that has ConsumerOptions.

Reply:
Not true here, this ConsumerConfig is the actual JetStream Consumer configuration as returned by the create API, it doesnt have extra functionality it is what it is, if something else will be needed in the future it would be a different struct key